modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/utils.py

```diff
@@ -13,15 +13,141 @@
 
 """Utilities for internal use by the ``HdkOnNativeDataframe``."""
 
+from typing import Tuple, Union
+from functools import lru_cache
 from collections import OrderedDict
 
 import pandas as pd
+from pandas import Timestamp
 from pandas.core.arrays.arrow.extension_types import ArrowIntervalType
 
 import pyarrow as pa
 from pyarrow.types import is_dictionary
 
 from modin.error_message import ErrorMessage
+from modin.utils import MODIN_UNNAMED_SERIES_LABEL
+
+
+class ColNameCodec:
+    IDX_COL_NAME = "__index__"
+    ROWID_COL_NAME = "__rowid__"
+
+    _RESERVED_NAMES = (MODIN_UNNAMED_SERIES_LABEL, ROWID_COL_NAME)
+    _COL_TYPES = Union[str, int, float, Timestamp, None]
+    _COL_NAME_TYPE = Union[_COL_TYPES, Tuple[_COL_TYPES, ...]]
+
+    def _encode_tuple(values: Tuple[_COL_TYPES, ...]) -> str:  # noqa: GL08
```

Comment:
I would suggest some reshuffling for methods and attrs. First go attrs, second go methods.